cipher/src/block.rs

```diff
@@ -32,13 +28,13 @@ pub trait NewBlockCipher: Sized {
     type KeySize: ArrayLength<u8>;
 
     /// Create new block cipher instance from key with fixed size.
-    fn new(key: &Key<Self>) -> Self;
+    fn new(key: &BlockCipherKey<Self>) -> Self;
 
     /// Create new block cipher instance from key with variable size.
     ///
     /// Default implementation will accept only keys with length equal to
     /// `KeySize`, but some ciphers can accept range of key lengths.
-    fn new_varkey(key: &[u8]) -> Result<Self, InvalidLength> {
+    fn new_var(key: &[u8]) -> Result<Self, InvalidLength> {
```

Comment:
As long as we're bikeshedding this method name, I've thought it would be clearer if it had `slice` in the name, e.g. `new_from_slice` or `from_slice`.

Reply:
`new_from_slice` sounds good! `from_slice` is shorter, but I think will be more confusing, since it could be mistaken for a specialized version of the `From` trait. Also usually `from*` methods imply simple conversion, which is not the case here.

i wonder if for stream cipher and block modes we should use plural `new_from_slices`, since they accept two slices.

Reply:
@newpavlov sounds good